weatherman.py

```diff
@@ -0,0 +1,398 @@
+            print("Invalid column alias")
+        else:
+            min_temp_column = weather_data.get_column_data(col_name,
+                                                           month, year)
+            if min_temp_column != []:
+                min_index = np.argmin(min_temp_column)
+                min_temp_list.append([min_temp_column[min_index],
+                                      min_index + 1, month])
+
+        # Max humidity
+        col_name = weather_data.get_column_name_from_alias(
+                                "max_humidity")
+        if col_name is None:
+            print("Invalid column alias")
+        else:
+            max_humidity_column = weather_data.get_column_data(col_name,
+                                                               month, year)
+            if max_humidity_column != []:
+                max_index = np.argmax(max_humidity_column)
+                max_humidity_list.append([max_humidity_column[max_index],
+                                          max_index + 1, month])
+
+        # Min humidity
+        col_name = weather_data.get_column_name_from_alias(
+                                "min_humidity")
+        if col_name is None:
+            print("Invalid column alias")
+        else:
+            min_humidity_column = weather_data.get_column_data(col_name,
+                                                               month, year)
+            if min_humidity_column != []:
+                min_index = np.argmin(min_humidity_column)
+                min_humidity_list.append([min_humidity_column[min_index],
+                                          min_index + 1, month])
+
+    # Checking largest/smallest in each of the lists
+    if len(max_temp_list) > 1:
+        max_temp_list = sorted(max_temp_list, reverse=True,
+                               key=lambda x: x[0])
+        print("Maximum Temperature:", max_temp_list[0][0], "C on",
+              get_month_string_from_number(max_temp_list[0][2]),
+              max_temp_list[0][1])
+
+    if len(min_temp_list) > 1:
+        min_temp_list = sorted(min_temp_list, key=lambda x: x[0])
+        print("Minimum Temperature:", min_temp_list[0][0], "C on"
+              get_month_string_from_number(min_temp_list[0][2]),
+              min_temp_list[0][1])
+
+    if len(max_humidity_list) > 1:
+        max_humidity_list = sorted(max_humidity_list, reverse=True,
+                                   key=lambda x: x[0])
+        print("Maximum Humidity:", max_humidity_list[0][0], "% on",
+              get_month_string_from_number(max_humidity_list[0][2]),
+              max_humidity_list[0][1])
+
+    if len(min_humidity_list) > 1:
+        min_humidity_list = sorted(min_humidity_list,
+                                   key=lambda x: x[0])
+        print("Minimum Humidity:", min_humidity_list[0][0], "% on",
+              get_month_string_from_number(min_humidity_list[0][2]),
+              min_humidity_list[0][1])
```

Comment:
all this can also be moved to a method. Maybe have 2 methods, `get_temperature_range` that returns highest and lowest temperature and `get_humidity_range` that returns highest and lowest humidity.